mlir/include/mlir/Dialect/DLTI/DLTI.h

```diff
@@ -103,6 +103,12 @@ class DataLayoutSpecAttr
   /// Returns the alloca memory space identifier.
   StringAttr getAllocaMemorySpaceIdentifier(MLIRContext *context) const;
 
+  /// Returns the program memory space identifier.
+  StringAttr getProgramMemorySpaceIdentifier(MLIRContext *context) const;
+
+  /// Returns the global memory space identifier.
+  StringAttr getGlobalMemorySpaceIdentifier(MLIRContext *context) const;
```

Comment:
Is there a doc that specifies what do these means? (The comments are just paraphrasing the API name right now)

Reply:
This is the data layout document in LLVM which describes these segments of the layout in a little more detail: https://llvm.org/docs/LangRef.html#data-layout 

I am happy to open a PR to add further comments if you wish however! But yes, I mimicked the comments from the original PR this one is based on which does the same.